thorlcr/thorutil/thmem.cpp

```diff
@@ -2315,19 +2316,22 @@ class CThorAllocator : public CSimpleInterface, implements IThorAllocator, imple
             rowManager->setReleaseWhenModifyCallback(true, true);
         }
     }
-    CThorAllocator(IThorAllocator &sharedAllocator, unsigned channel)
+    CThorAllocator(CThorAllocator &sharedAllocator, unsigned channel)
     {
-        allocatorMetaCache.setown(createRowAllocatorCache(this));
+        allocatorMetaCache.set(sharedAllocator.queryAllocateCache());
         rowManager.set(sharedAllocator.queryRowManager()->querySlaveRowManager(channel));
         defaultFlags = sharedAllocator.queryFlags();
         logctx = sharedAllocator.queryLoggingContext();
         numChannels = 0;
+        dbgassertex(channel <= 0xffff);
```

Comment:
Whoops, yes would need to be 0xff - and that is low.

I was initially going to change the activityId to an unsigned __int64, and use the upper 32bits fo channel.

Will change to that.